strawberry/printer.py

```diff
@@ -35,33 +52,95 @@
     from strawberry.schema import BaseSchema
 
 
-def print_schema_directive_params(params: Dict) -> str:
+_T = TypeVar("_T")
+
+
+@dataclasses.dataclass
+class PrintExtras:
+    directives: Set[str] = dataclasses.field(default_factory=set)
+    types: Set[type] = dataclasses.field(default_factory=set)
+
+
+@overload
+def _serialize_dataclasses(value: Dict[_T, object]) -> Dict[_T, object]:
+    ...
+
+
+@overload
+def _serialize_dataclasses(value: Union[List[object], Tuple[object]]) -> List[object]:
+    ...
+
+
+@overload
+def _serialize_dataclasses(value: object) -> object:
+    ...
+
+
+def _serialize_dataclasses(value):
+    if dataclasses.is_dataclass(value):
+        return dataclasses.asdict(value)
+    if isinstance(value, (list, tuple)):
+        return [_serialize_dataclasses(v) for v in value]
+    if isinstance(value, dict):
+        return {k: _serialize_dataclasses(v) for k, v in value.items()}
+
+    return value
+
+
+def print_schema_directive_params(
+    directive: GraphQLDirective, values: Dict[str, Any]
+) -> str:
+    params = []
+    for name, arg in directive.args.items():
+        value = values.get(name, arg.default_value)
+        if value is UNSET:
+            value = None
+        else:
+            ast = ast_from_value(_serialize_dataclasses(value), arg.type)
+            value = ast and f"{name}: {print_ast(ast)}"
+
+        if value:
+            params.append(value)
+
     if not params:
         return ""
 
-    return (
-        "("
-        + ", ".join(
-            f'{name}: "{value}"' for name, value in params.items() if value is not UNSET
-        )
-        + ")"
-    )
+    return "(" + ", ".join(params) + ")"
 
 
-def print_schema_directive(directive: Any, schema: BaseSchema) -> str:
-    name_converter = schema.config.name_converter
+def print_schema_directive(
+    directive: Any, schema: BaseSchema, *, extras: PrintExtras
+) -> str:
+    strawberry_directive = cast(
+        StrawberrySchemaDirective, directive.__class__.__strawberry_directive__
+    )
+    schema_converter = schema.schema_converter
+    gql_directive = schema_converter.from_schema_directive(directive)
```

Comment:
in theory the printer shouldn't access the schema converter (it shouldn't know about it)

do you think we change this code to to use the directive directly?

Reply:
Sure! I didn't know if I should make a big change here so I went for this

We can probably extract and save the schema directives in the `Schema` itself when creating it, just like we do with types, client directives, etc and then retrieve it from there.

Will do that and send an update in this PR soon.

obs. this will also fix the other comment you've made since there will be no reason to access the `schema_converter` in here